### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by anapdh**
This branch features:

- 3 Models and controllers, one for the User, the Event and the EventAttendances;
- The belong_to and has_many properly implemented to create associations between each model and its tables;
- Validations for Events controller;
- A sign in, sign up and, when logged in, New Event creation;
- Each user can create events and attend upcoming events when clicking on the Show page for the event (past events can't be marked as attending);
- The Events registers the user that has created them;
- All created events displays on the root page, separated by upcoming and past events;
- On the User page (clicking on its name), the events he attended or will attend shows up;
- Upcoming and past events refactored into scopes (PS: the original code is commented out in order to show that this change was made);
- Helper methods for removing logic from views (PS: the project specifies that inside user/show.html.erb is allowed to have logic).

### CODE REVIEW

File: app/models/event.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  scope :upcoming, -> { where('date > ?', Time.now) }
+  scope :past, -> { where('date < ?', Time.now) }
+
+  # def self.upcoming
+  #   self.where("date > ?", Time.now)
+  # end
+
+  # def self.past
+  #   self.where("date < ?", Time.now)
+  # end
```
Comment (kapkev) [inline app/models/event.rb]: - [x] Please remove commented code 😄

**Review by kapkev — CHANGES_REQUESTED** [review]
# [STATUS] CHANGES REQUIRED♻️

#### Boa Noite @anapdh <3 (long time no see) e @jpdf00 ! _As project always looks amazing, and you make our job very easy._ However, there's a silly thing that I'm sure you both just forgot to remove it, and to accomplish the actual project, you are required to only one minor change:
 
## [REQUIRED CHANGE]
- [x] Please remove the commented code at events model;

## [OPTIONAL CHANGE]
- [x] I'm being able to destroy an event even if I'm not logged in.

### Please make the required change and resubmit it for a new CR 💪

#### First time reviewing a fully Brazilian project here at Microverse 🚀


👨🏻‍💻 **TSE: Kevin Alves**

  

📫 Have any question? Better call Kevin:

<a  href="mailto:kevinalves.js@gmail.com"> gmail</a> [Twitter Follow](https://twitter.com/intent/follow?screen_name=kevinnnmn) [](https://microverse-students.slack.com/team/UT86D3AP4)

<sub>*We're here to help you, to ensure that you're using the best practices and getting you ready for the job market. In the real world, you'll have many change requests for your real clients; it's good to get used to it.</sub>

**Comment by kapkev** [comment]
### Vai Brasil!

**Review by acushlakoncept — APPROVED** [review]
# Status: PROJECT APPROVED  🎉 🚀

### Hi, @anapdh and @jpdf00 excellent job 👍 👏 👏  both of you have done an impressive job. Congratulations on completing your project. 🍾 

I will go ahead and mark this as complete, feel free to merge when you are ready. All the best with the next project.

### Cheers and Happy coding! 💻 👏

(Feel free to contact me on slack(@acushla), if you have any additional question)
Uduak Essien
[Twitter](https://twitter.com/acushlakoncept) | [LinkedIn](https://www.linkedin.com/in/acushlakoncept/)

**Pull request merged**